cli/defenseclaw/observability/v8_migration.py

```
@@ -2629,15 +2765,20 @@ def _endpoint_host(endpoint: str, ctx: _Context, path: str) -> str:
     try:
         parsed = urlsplit(endpoint if "://" in endpoint else f"//{endpoint}")
         host = parsed.hostname
+        has_inline_credentials = parsed.username is not None
+        # urllib defers malformed and out-of-range port errors until access.
+        # Force that check here so callers always receive a bounded migration
+        # diagnostic instead of a later generic candidate-validation failure.
+        parsed.port
```

Ruff (0.15.21)

[warning] Found useless expression. Either assign it to a variable or remove it.

(B018)